source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -516,10 +541,39 @@ namespace
         }
     }
 
-    void writeDocument(DocumentPtr& doc, PrvStage* stage, RtFileIo::WriteFilter filter)
+    void writeSourceUris(PrvStage* stage, DocumentPtr doc)
+    {
+        const PrvObjectHandleVec& refs = stage->getReferencedStages();
+        for (size_t i = 0; i < refs.size(); i++)
+        {
+            const PrvObjectHandle refStage = refs[i];
+            PrvStage* pStage = refStage->asA<PrvStage>();
+            if (pStage->numReferences())
+            {
+                writeSourceUris(pStage, doc);
```

Comment:
Not sure but should we write includes from referenced stages into the document for this stage?